### PR TITLE
Enable use of image stanza in `pyflyte register` and other small changes

### PR DESCRIPTION
**Pull request by eapolinario**
# TL;DR
`pyflyte register` is now able to load images present in the config file. That sub-command also requires a version in non-fast case.

## Type
 - [x] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [x] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
Changes to `pyflyte register`:
1. The output of the command now is more explicit about which serialization settings are used, which include the images allowed to be used.
2. Force version in non fast. The idea is to make it clear that this option requires the user to be deliberate about versions when registering entities using the non-fast option.
3. Allow for images present in a config file to be used during registration

## Tracking Issue
https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: flytekit/clis/sdk_in_container/register.py [hunk 1]
```diff
@@ -194,5 +202,9 @@ def register(
         version = remote._version_from_hash(md5_bytes, serialization_settings, service_account, raw_data_prefix)  # noqa
         cli_logger.info(f"Computed version is {version}")
 
+    cli_logger.warn(
```
Comment (wild-endeavor) [inline flytekit/clis/sdk_in_container/register.py:205]: can we use info or lower?  if this is something we want to print then click.echo?

**Review by wild-endeavor — APPROVED** [review]

File: flytekit/clis/sdk_in_container/register.py (hunk 1 above)
Comment (eapolinario) [reply flytekit/clis/sdk_in_container/register.py:205]: this should be printed. I'm going to change it to `click.echo`.

**Review by wild-endeavor — APPROVED** [review]
